### ISSUE TITLE
 [AC] [Feature] [Group/Core] Bot Gear Exceptions Table

### ISSUE THREAD

**Issue by Day36512** [trickerer/Trinity-Bots#518]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that my issue may get **closed without notice** if I *intentionally* remove or skip any mandatory\* field
- [X] This is NOT a bug/crash report
- [X] I've searched the bugtracker for similar questions/requests **including closed ones**

### Description

Hey Trickerer,

I was wondering if we could have an sql table that we could add item IDs for exceptions for random bots not to equip or pick up. That way we can limit some of the useless npcitems and maybe custom transmog items bots might have. Or any god items people may have created. This would be something the owner would do themself since I'm not expecting you to put in any item IDs into the table.

Thanks!

### TrinityCore or AzerothCore

Acore

### Core rev. hash/commit

[6768dda08d]

### Operating system

Windows

**Comment by trickerer** [comment]
Answering in order.
1) Equipment for wandering bots requires a more elaborate solution: white lists of items per class per slot per level bracket. Eventually I'll prepare a couple commands to make it easier for myself and implement the feature.
2) Blacklisting items allowed for **owned** bots to equip is possible although I feel kinda skeptical about this idea. I imagine admin or gm giving trying to explain to an agitated player about how this works and why they've disallowed some items. Probably needs some whisper from bot when this happens.
3) To prevent any item from being used for transmog defeats the purpose of the feature I fear. I'm not going for it.

Overall this is 2 new tables in DB.

**Comment by Day36512** [comment]
Sorry I should clarify. This would only disable wandering bot equipment for picking up useless items, not affect player npcbots. Disabling items like this one for example: https://wowgaming.altervista.org/aowow/?item=7612

If you made such a table I'd be more than happy to share the entries for just such items.

**Comment by Day36512** [comment]
And custom transmog items i was referring to was, for example, along the same lines as the Monster - type items where they have no stats but are used for tmog only. But yeah not wanting to disable tmog, just was an example of removing useless items from pvp/wandering bots.

**Comment by trickerer** [comment]
Just one table then. `2` and `3` can be discarded.

**Comment by Day36512** [comment]
If you need help creating such a table, just let me know what structure you'd like and I can start making queries and filling it out. No rush of course.

**Comment by trickerer** [comment]
I've added a command to list leveled generated gear per class per slot and a checker after generation. As one could expect, this uncovered a ton of bugs in resulting gear sets.
A table to check for disabled ids is also there. After looking at just how many items are generated in total for all 20 classes I decided to use blacklisting approach afterall.

Oh, and remember that walking ad board or whatever item? I found it. It's a cape called Doomsday Message. Disabled it.

If you want to use gear list command you'll need to uncomment npcbot debug root command (in botcommands.cpp). I'd suggest using some chat addon to increase the number of chat history lines. Like `Prat` or something.
